core/src/stored/reserve.cc

```diff
@@ -427,7 +427,10 @@ bool FindSuitableDeviceForJob(JobControlRecord* jcr, ReserveContext& rctx)
 
       // Check with Director if this Volume is OK
       bstrncpy(dcr->VolumeName, vol->vol_name, sizeof(dcr->VolumeName));
-      if (!dcr->DirGetVolumeInfo(GET_VOL_INFO_FOR_WRITE)) { continue; }
+      if (!dcr->DirGetVolumeInfo(GET_VOL_INFO_FOR_WRITE)) {
+        dcr->VolumeName[0] = 0;
+        continue;
```

Comment:
What happens if `DirGetVolumeInfo` succeeds but we do not find a usable device that can write to that volume ? In that case the volume name should also be reset since we will not use that volume!

Reply:
I would suggest reverting this change and instead doing
```
  if (ok) {
    Dmsg1(debuglevel, "OK dev found. Vol=%s from in-use vols list\n",
          rctx.VolumeName);
    return true;
  } else {
    // no existing volume found; reseting volume name.
    dcr->VolumeName[0] = '\0';
  }
```
after the loop.